packages/phoenix-event-display/src/tests/managers/ui-manager/phoenix-menu/phoenix-menu-ui.test.ts

```diff
@@ -0,0 +1,155 @@
+
+jest.mock('../../../../managers/three-manager');
+
+describe('PhoenixMenuUI', () => {
+  let phoenixMenuUI: PhoenixMenuUI;
+  let phoenixMenuUIPrivate: any;
+  let phoenixMenuRoot: PhoenixMenuNode;
+  let three: ThreeManager;
+
+  beforeEach(() => {
+    three = new ThreeManager(new InfoLogger());
+    phoenixMenuRoot = new PhoenixMenuNode('test');
+    phoenixMenuUI = new PhoenixMenuUI(phoenixMenuRoot, three);
+    phoenixMenuUIPrivate = phoenixMenuUI as any;
+  });
+
+  afterEach(() => {
+    phoenixMenuUI.clear();
+  });
+
+  it('should create an instance', () => {
+    expect(phoenixMenuUI).toBeTruthy();
+  });
+
+  it('should add geometry (detector geometry) folder to the menu', () => {
+    phoenixMenuUI.addGeometryFolder();
+
+    expect(phoenixMenuUIPrivate.geomFolder.name).toBe('Detector');
+    expect(phoenixMenuUIPrivate.geomFolder).toBeInstanceOf(PhoenixMenuNode);
+  });
+
+  it('should add event data folder with functions for event data toggles like show/hide and depthTest', () => {
+    phoenixMenuUI.addEventDataFolder();
+
+    expect(phoenixMenuUIPrivate.eventFolder.name).toBe('Event Data');
+    expect(phoenixMenuUIPrivate.eventFolder).toBeInstanceOf(PhoenixMenuNode);
+  });
+
+  it('should add labels folder to the menu', () => {
+    phoenixMenuUI.addLabelsFolder('test');
+
+    expect(phoenixMenuUIPrivate.labelsFolder.name).toBe('Labels');
+    expect(phoenixMenuUIPrivate.labelsFolder).toBeInstanceOf(PhoenixMenuNode);
+  });
+
+  it('should remove label from the menu and scene if it exists', () => {
+    const testMenuNode = new PhoenixMenuNode('test');
+    const testMenuNodeChild = testMenuNode.addChild('test', () => {});
+
+    jest.spyOn(testMenuNodeChild, 'remove');
+
+    phoenixMenuUI.removeLabel('test', testMenuNodeChild);
+
+    expect(testMenuNodeChild.remove).toHaveBeenCalled();
+  });
+
+  it('should add geometry to the menu geometry folder and set up its configurable options', () => {
+    phoenixMenuUI.addGeometryFolder();
+    phoenixMenuUI.addGeometry(new Object3D(), 'test');
+
+    expect(phoenixMenuUIPrivate.geomFolder.children[0].name).toBe('test');
+    expect(phoenixMenuUIPrivate.geomFolder.parent.name).toBe('test');
+  });
+
+  it('should add folder for event data type like tracks or hits to the menu', () => {
+    phoenixMenuUI.addEventDataFolder();
+    phoenixMenuUI.addEventDataTypeFolder('Event Data');
+
+    expect(phoenixMenuUIPrivate.eventFolder.children[0]).toBeInstanceOf(
+      PhoenixMenuNode
+    );
+    expect(phoenixMenuUIPrivate.eventFolder.children[0].name).toBe(
+      'Event Data'
+    );
+  });
+
+  it('should add collection folder and its configurable options to the event data type (tracks, hits etc.) folder', () => {
+    phoenixMenuUI.addEventDataFolder();
+    phoenixMenuUI.addEventDataTypeFolder('test');
+
+    jest.spyOn(three, 'getColorManager');
+
+    phoenixMenuUI.addCollection('test', 'Tracks');
+
+    expect(three.getColorManager).toHaveBeenCalled();
```

Comment:
This doesn't check if the collection folder/node was added.

https://github.com/HSF/phoenix/blob/master/packages/phoenix-event-display/src/managers/ui-manager/phoenix-menu/phoenix-menu-ui.ts#L222

Reply:
I wanted to spy on typeFolder.addChild but I don't understand why it does not execute further from here in this function where we find out the collectionNode - 
https://github.com/HSF/phoenix/blob/5ec4ed3fb8818144530e56a5e5fcb4c46ec54557/packages/phoenix-event-display/src/managers/ui-manager/phoenix-menu/phoenix-menu-ui.ts?rgh-link-date=2022-07-13T18%3A25%3A10Z#L224

    		const collectionNode = typeFolder.addChild(
     		  collectionName,
      >>		  (value: boolean) => {
      >>		    const collectionObject = this.sceneManager
       >>   		   .getObjectByName(SceneManager.EVENT_DATA_ID)
       >>   		   .getObjectByName(collectionName);
       >>		 this.sceneManager.objectVisibility(collectionObject, value);
       >>	    }
   			  );
Any idea why? 😅

Reply:
Oh, that makes sense. That is a function which only executes on toggle.

Instead of spying on the `addChild`, you could check if the added child is in the tree by using `typeFolder.findInTree`.

Reply:
cool!